test/e2e/all_in_one_test.go

```diff
@@ -325,6 +326,19 @@ func TestDeployAllInOneDBLESS(t *testing.T) {
 	t.Log("scale proxy to 3 replicas and wait for all instances to be ready")
 	scaleDeployment(ctx, t, env, deployments.ProxyNN, 3)
 	ensureAllProxyReplicasAreConfigured(ctx, t, env, deployments.ProxyNN)
+
+	t.Log("scale proxy to 1 replica")
+	scaleDeployment(ctx, t, env, deployments.ProxyNN, 1)
+
+	t.Log("misconfigure the ingress")
+	reconfigureExistingIngress(ctx, t, env, ingress, func(i *netv1.Ingress) {
+		ingress.Spec.Rules[0].HTTP.Paths[0].Path = badEchoPath
+	})
+
+	t.Log("scale proxy to 3 replicas and verify that the new replicas get the old good configuration")
+	scaleDeployment(ctx, t, env, deployments.ProxyNN, 3)
+	// verify all the proxy replicas have the last good configuration
+	ensureAllProxyReplicasAreConfigured(ctx, t, env, deployments.ProxyNN)
```

Comment:
As covering this feature on the E2E level totally makes sense, I'd also suggest adding a test for `KongClient` that would verify this on a unit level (`TestKongClientUpdate_*` can serve as an example for mocking `KongClients` dependencies).